core/src/ast/methods.rs

```diff
@@ -94,6 +94,23 @@ impl Method {
             .iter()
             .for_each(|t| t.check_opaque(env, errors));
     }
+
+    /// Checks whether the method qualifies for special writeable handling.
+    /// To qualify, a method must:
+    ///  - not return any value
+    ///  - have the last argument be an `&mut diplomat_runtime::DiplomatWriteable`
+    ///
+    /// Typically, methods of this form will be transformed in the bindings to a
+    /// method that doesn't take the writeable as an argument but instead creates
+    /// one locally and just returns the final string.
+    pub fn is_writeable_out(&self) -> bool {
+        // TODO(shadaj): support results with empty success value
+        // TODO(shadaj): reconsider if we should auto-detect writeables
```

Comment:
thought: Perhaps we should be panicking if we have any writeable type params and they are _not_ the last param (or if there is a return type)

Reply:
Right now, this will effectively happen since we'll give up on coming up with a way to take the `Writeable` as a parameter in the FFI declarations. I feel like we should just fall back to a manual `Writeable` API when the conditions aren't met (or alternatively just require an attribute opting-into the special behavior and panic when the signature isn't right).

Reply:
Ah that works